ykrt/pass_finder/try_passes.py

```diff
@@ -66,53 +66,79 @@ def log(logf, s):
 def executable_exists(cmd):
     return shutil.which(cmd) is not None
 
-def get_all_passes(is_prelink):
-    cmd = get_opt_cmd(is_prelink) 
-    p = Popen(cmd, shell=True, stdout=PIPE, close_fds=True)
-    sout, serr = p.communicate() 
-    assert(p.returncode == 0)
+def split_passes(passes_string):
+    parts = []           
+    temp_part = []      
+    paren_stack = []     
```

Comment:
I think `paren_stack` is a bool in disguise. At most it's an interger in disguise because the values stored in this list aren't checked. Personally I would just have an integer `nesting = 0` and inc/dec it for both types of brackets. If someone gives an invalid string (e.g. switches closed brackets around), this will not work properly (but nor does the current code), but that can come under the "someone else shouldn't do that" category :)

Reply:
Addressed here: 0ebf4e6be0f59bb4b28fc16e91db87d6dcbe50f7

Reply:
https://github.com/ykjit/yk/commit/0ebf4e6be0f59bb4b28fc16e91db87d6dcbe50f7 is better but i think you can simplify this to just `nesting` --- you don't gain anything at the moment from checking the nesting of parens and angle brackets separately.

Reply:
Addressed here: d2c78c9eba0a7d94101a39cef66404217c8e5306